packages/grid/x-data-grid/src/utils/createSelector.ts

```diff
@@ -86,6 +86,40 @@ export const createSelector: CreateSelectorFunction = (...args: any) => {
   return selector;
 };
 
+export const createRawSelector: CreateSelectorFunction = ((
+  a: Function,
+  b?: Function,
+  c?: Function,
+  d?: Function,
+  ...rest: any[]
+) => {
+  if (rest.length > 0) throw new Error('Unsupported number of selectors');
+
+  let selector: any;
+
+  if (a && b && c && d) {
+    selector = (stateOrApiRef: any) => {
+      return d(c(b(a(stateOrApiRef.current ? stateOrApiRef.current.state : stateOrApiRef))));
+    };
+  } else if (a && b && c) {
+    selector = (stateOrApiRef: any) => {
+      return c(b(a(stateOrApiRef.current ? stateOrApiRef.current.state : stateOrApiRef)));
+    };
+  } else if (a && b) {
+    selector = (stateOrApiRef: any) => {
+      return b(a(stateOrApiRef.current ? stateOrApiRef.current.state : stateOrApiRef));
+    };
+  } else {
+    throw new Error('Missing arguments');
+  }
+
+  // We use this property to detect if the selector was created with createSelector
+  // or it's only a simple function the receives the state and returns part of it.
+  selector.acceptsApiRef = true;
+
+  return selector;
+}) as unknown as CreateSelectorFunction;
+
```

Comment:
I've specialized the selectors for their arguments so we avoid looping. I plan to rename `createRawSelector` to `createSelector` and make it the default in the future, with `createSelectorMemoized` being used sparingly if needed. But I'll keep that for another PR, this one is big enough already.